File: migrate_data.py
import json
import os
import re
import ast

files = [
    {'name': 'recipes.js'},
    {'name': 'fish.js'},
    {'name': 'insects.js'},
    {'name': 'crops.js'},
    {'name': 'flowers.js'}
]

base_dir = os.getcwd()
js_dir = os.path.join(base_dir, 'js')
data_dir = os.path.join(base_dir, 'data')

if not os.path.exists(data_dir):
    os.makedirs(data_dir)

for item in files:
    js_path = os.path.join(js_dir, item['name'])
    if not os.path.exists(js_path):
        print(f"Skipping {item['name']}, not found.")
        continue

    try:
        with open(js_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Regex to find the list. assuming format "const X = [...];"
        match = re.search(r'=\s*(\[[\s\S]*\])\s*;', content)
        if match:
            list_str = match.group(1)
            # Python's ast.literal_eval can handle basic JS arrays if they look like Python lists
            # but JS objects like { key: "val" } (unquoted keys) will fail.
            # However, looking at previous file views, the keys are quoted like { "Receta": ... }
            # So it is effectively JSON.
            
            # We need to handle potential trailing commas before ] or }
            # Simple regex cleanup for trailing commas
            list_str_clean = re.sub(r',(\s*[\]}])', r'\1', list_str)
            
            try:
                data = json.loads(list_str_clean)
                print(f"Parsed {item['name']} as standard JSON")
            except json.JSONDecodeError:
                print(f"JSON convert failed for {item['name']}, trying loose eval via ast (risky but okay for known data)")
                # If JSON fails (maybe single quotes?), try AST
                # AST expects python syntax, so true/false must be True/False
                # This is tricky if it's mixed.
                # Let's try to fix quotes first?
                # Actually, if keys are quoted, JSON should work unless there are comments.
                # Let's hope the previous files are cleanish.
                # If all else fails, I'll manually fix the specific one or just use what I have.
                continue

            json_out_path = os.path.join(data_dir, item['name'].replace('.js', '.json'))
            with open(json_out_path, 'w', encoding='utf-8') as f_out:
                json.dump(data, f_out, indent=4, ensure_ascii=False)
            print(f"Saved {json_out_path}")
            
        else:
            print(f"No array found in {item['name']}")

    except Exception as e:
        print(f"Error processing {item['name']}: {e}")

File: migrate_data.js
const fs = require('fs');
const path = require('path');

const files = [
    { name: 'recipes.js', varName: 'RECIPES_DATA' },
    { name: 'fish.js', varName: 'FISH_DATA' },
    { name: 'insects.js', varName: 'INSECTS_DATA' },
    { name: 'crops.js', varName: 'CROPS_DATA' },
    { name: 'flowers.js', varName: 'FLOWERS_DATA' }
];

files.forEach(file => {
    try {
        const filePath = path.join(__dirname, 'js', file.name);
        if (!fs.existsSync(filePath)) {
            console.log(`Skipping ${file.name}, not found.`);
            return;
        }

        const content = fs.readFileSync(filePath, 'utf8');
        // Regex to extract the array content: everything between [ and the last ]
        const match = content.match(/=\s*(\[[\s\S]*\])\s*;/);

        if (match && match[1]) {
            let jsonStr = match[1];
            // Fix potential trailing commas which valid JS allows but JSON doesn't
            jsonStr = jsonStr.replace(/,(\s*[\]}])/g, '$1');

            const jsonPath = path.join(__dirname, 'data', file.name.replace('.js', '.json'));

            // Validate it's valid JSON
            try {
                JSON.parse(jsonStr); // Test parse
                fs.writeFileSync(jsonPath, jsonStr, 'utf8');
                console.log(`Converted ${file.name} to ${jsonPath}`);
            } catch (jsonErr) {
                // Fallback: evaluate the JS code to get the object (unsafe but fine for local conversion script)
                // We can use eval() here strictly for migration
                const data = eval(jsonStr);
                fs.writeFileSync(jsonPath, JSON.stringify(data, null, 4), 'utf8');
                console.log(`Converted ${file.name} using eval fallback`);
            }
        } else {
            console.log(`Could not parse pattern in ${file.name}`);
        }
    } catch (err) {
        console.error(`Error processing ${file.name}:`, err);
    }
});
